modules/core/keeper/msg_server.go

```diff
@@ -612,15 +616,14 @@ func (k Keeper) Acknowledgement(goCtx context.Context, msg *channeltypes.MsgAckn
 	cacheCtx, writeFn := ctx.CacheContext()
 	err = k.ChannelKeeper.AcknowledgePacket(cacheCtx, cap, msg.Packet, msg.Acknowledgement, msg.ProofAcked, msg.ProofHeight)
 
-	// NOTE: The context returned by CacheContext() refers to a new EventManager, so it needs to explicitly set events to the original context.
-	ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
-
 	switch err {
 	case nil:
 		writeFn()
 	case channeltypes.ErrNoOpMsg:
+		// no-ops do not need event emission as they will be ignored.
```

Comment:
```suggestion
		// no-ops do not need event emission as they will be ignored
```
Typically I only add periods if the comment is structured as a valid english sentence (begins with capital, etc). In this situation I think it makes sense to keep it informal